twisted/internet/inotify.py

```diff
@@ -137,6 +137,7 @@ def _notify(self, filepath, events):
         """
         Callback function used by L{INotify} to dispatch an event.
         """
+        filepath = filepath.asBytesMode()
```

Comment:
Cool, this is fun!

So, this line here reveals a further question. Here we convert the filepath unconditionally into a bytes mode path before passing it to the user. That raises two questions:

1. Is this actually necessary? As far as I can tell we only have to do this because we convert the path into bytes mode in the constructor to `_Watch`. However, if we converted the path to bytes mode in `INotify.watch` instead (with a similar conversion in`Inotify.ignore`), we'd probably avoid needing to do this because the entire thing would be in bytes mode the whole time.

2. *However*, that then raises a follow-on question. Is it surprising to provide the user a bytes-mode filepath in the callback when they gave us a text-mode one? I'm genuinely not sure about the answer to this: your feedback would be interesting, as would @adiroiban's.